geps/gep-713/index.md

```diff
@@ -1,1157 +1,1094 @@
 
+  ```go
+  package color
 
-## Naming Policy objects
+  import (
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    gatewayapiv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+  )
 
-Although Direct and Inherited Policies behave differently in many respects, in
-general they should be named using similar rules.
+  type ColorPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
 
-Policy objects MUST be clearly named so as to indicate that they are Policy
-metaresources.
+    // Spec defines the desired state of the policy.
+    Spec ColorPolicySpec `json:"spec"`
 
-The simplest way to do that is to ensure that the type's name contains the `Policy`
-string.
+    // Status defines the current state of the policy.
+    Status ColorPolicyStatus `json:"status,omitempty"`
+  }
 
-Implementations SHOULD use `Policy` as the last part of the names of object types
-that use this pattern.
+  type ColorPolicySpec struct {
+    // TargetRefs specify the targets of the policy by name.
+    // The following kinds are supported: …
+    // +listType=map
+    // +listMapKey=group
+    // +listMapKey=kind
+    // +listMapKey=namespace
+    // +listMapKey=name
+    // +kubebuilder:validation:MinItems=1
+    // +kubebuilder:validation:MaxItems=16
+    TargetRefs []gatewayapiv1alpha2.NamespacedPolicyTargetReference `json:"targetRefs"`
 
-If an implementation does not, then they MUST clearly document what objects
-are Policy metaresources in their documentation. Again, this is _not recommended_
-without a _very_ good reason.
+    // rest of the spec ("spec proper")…
+  }
+  ```
+</details>
 
-### Targeting Virtual Types
-In some cases (likely limited to mesh) we may want to apply policies to requests
-to external services. To accomplish this, implementations MAY choose to support
-a reference to a virtual resource type. For example:
+#### Spanning behavior across relationships of a target
+
+Because the target objects that give a metaresource context sometimes are themselves inserted into a broader context of their own, composed of other interrelated objects, the effects of a metaresource can be limited to a target object itself or span across the links between this object and other objects that the object is related to.
+
+E.g. – a metaresource that targets a Namespace may declare intent that affects the behavior of the namespace itself (for what concerns to the implementation of Namespaces in Kubernetes) or alternatively it can act as a means to affect the behavior of other objects that exist in the referred namespace (e.g. ConfigMaps). While in the former case, the (direct) target object is the Namespace itself, in the latter the (indirect) target is a set of objects of a different kind (e.g. ConfigMaps.)
```

Comment:
Instead of direct and indirect, we could still use the terms associated with scope, e.g. local and global.  A global reference has sub-objects as targets, and local reference stands alone as the target.